### PR TITLE
[Color system] implement color system in button round 2

### PR DESCRIPTION
**Pull request by danrosenthal**
## Supplementary states for tophatting

```tsx
<Stack>
  <Button plain destructive icon={ExternalMinor}>
    Add product
  </Button>
  <Button plain destructive disabled icon={ExternalMinor}>
    Add product
  </Button>
  <Button plain disabled icon={ExternalMinor}>
    Add product
  </Button>
  <Button plain icon={ExternalMinor} />
  <Button plain disabled icon={ExternalMinor} />
  <Button plain destructive icon={ExternalMinor} />
  <Button plain disabled destructive icon={ExternalMinor} />
  <Button disabled destructive icon={ExternalMinor}>
    Add product
  </Button>
  <Button disabled icon={ExternalMinor}>
    Add product
  </Button>
  <Button outline destructive disabled>
    Add product
  </Button>
  <Button outline disabled>
    Add product
  </Button>
</Stack>
```

### CODE REVIEW

**Comment by danrosenthal** [comment]
☝️ rebased to remove in progress from commit

File: src/components/Button/Button.scss [hunk 1]
```diff
@@ -243,58 +246,30 @@ $partial-button-filled-pressed-box-shadow: inset 0 0 0 0 transparent,
...
   &:hover,
   &:focus,
   &:active {
+    @include recolor-icon(color('blue', 'dark'));
     background: transparent;
     border: 0;
     box-shadow: none;
   }
 
-  // TODO: implement depressed state for global theming
```
Comment (danrosenthal) [inline src/components/Button/Button.scss]: This still needs to be implemented, but designs are TBD

**Comment by tmlayton** [comment]
Are these the right styles for outline buttons?

Screen Shot 2019-10-28 at 4 33 00 PM

If so, the focus ring will need tweaking in this case so the ring does not sit on top of and partially overlay the button border.

**Comment by danrosenthal** [comment]
Also FYI, this is my design reference if you want to double check while tophatting: https://www.figma.com/file/4dAAt5iFPSaxUKiYVKrkYj/MB%E2%80%93Polaris-(Web)%3A-Components?node-id=7%3A3

**Pull request merged**